tests/ui/implicit_saturating_sub.fixed

```diff
@@ -222,3 +222,14 @@ fn main() {
         a - b
     };
 }
+
+// https://github.com/rust-lang/rust-clippy/issues/13524
+fn regression_13524(a: usize, b: usize, c: bool) -> usize {
+    if c {
+        123
+    } else if a >= b {
+        b.saturating_sub(a)
+    } else {
+        b - a
+    }
```

Comment:
I'm a bit late, but reading through #13524, shouldn't the fixed code be just `} else { b.saturating_sub(a) }` (old suggestion was almost correct but it was missing braces)?

The new suggestion adds more code instead of simplifying it like before (changes a `0` to `saturating_sub` instead of the whole `if`)? 👀

Reply:
I did some more debugging, and yep, it was a mistake, #13542 is now open reverting this and implementing a more robust solution :heart: